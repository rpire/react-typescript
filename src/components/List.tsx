import { FunctionComponent } from 'react';

type PropType = {
  arr: string[],
}

const List: FunctionComponent<PropType> = ({ arr }) => (
  <ol>
    {arr.map((element) => (
      <li key={element}>{element}</li>
    ))}
  </ol>
);

export default List;
