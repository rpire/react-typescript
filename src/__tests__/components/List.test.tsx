import * as ReactDom from 'react-dom';
import List from '../../components/List';

describe('List component tests', () => {
  let container: HTMLDivElement;

  const arr: string[] = ['foo', 'boo', 'loo', 'goo', 'zoo'];

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);

    ReactDom.render(<List arr={arr} />, container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders every element in "arr"', () => {
    const listItems = container.querySelectorAll('li');

    expect(listItems).toHaveLength(arr.length);
    expect(listItems[0]).toContainHTML('foo');
    expect(listItems[2]).toContainHTML('loo');
    expect(listItems[4]).toContainHTML('zoo');
  });
});
