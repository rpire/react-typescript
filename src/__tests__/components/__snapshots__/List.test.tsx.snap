// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component tests renders correctly 1`] = `
<div>
  <ol>
    <li>
      foo
    </li>
    <li>
      boo
    </li>
    <li>
      loo
    </li>
    <li>
      goo
    </li>
    <li>
      zoo
    </li>
  </ol>
</div>
`;
